README.md

```diff
@@ -1,42 +1,44 @@
 # rubydeps
 
+The DependencyWorker currently performs the following actions:
+
+* Fetch gemfile and gemspec for a given Gem
+  * Currently, the GithubGem class specifies the needed information for a gem hosted on github. Other providers could be added.
+* Parse both files for dependency-requirements
+* Determine the latest version for each required gem
+* Generate a Hash of results of the following form:
+```ruby
+{
+  "gemspec" => [
+    {"name"=>"gollum-lib", "required"=>">= 4.2.10, ~> 4.2", "latest"=>"4.2.10", "outdated"=>false},
+    {"name"=>"kramdown", "required"=>"~> 1.9.0", "latest"=>"2.1.0", "outdated"=>true}
+  ],
+  "gemfile" => [
+    {"name"=>"rake", "required"=>"~> 10.4", "latest"=>"13.0.1", "outdated"=>true}
+  ]
+}
 ```
-$ bundle exec ruby test.rb 
-First trying with Gem::SpecFetcher
-Checking dependencies status for rake
-Current required version of rake: < 14.0, >= 10.0
-Latest version of rake: 13.0.1
-So is the rake requirement outdated? true
-
-Checking dependencies status for gollum
-Current required version of gollum: < 4.0, >= 2.0
-Latest version of gollum: 4.1.4
-So is the gollum requirement outdated? false
-
-Benchmark with Gem::SpecFetcher
-Rehearsal ----------------------------------------------
-             0.000000   0.000000   0.000000 (  0.001200)
-------------------------------------- total: 0.000000sec
-
-                 user     system      total        real
-             0.000000   0.000000   0.000000 (  0.000706)
-
-Now trying with rubygems API
-Checking dependencies status for rake
-Current required version of rake: < 14.0, >= 10.0
-Latest version of rake: 13.0.1
-So is the rake requirement outdated? true
-
-Checking dependencies status for gollum
-Current required version of gollum: < 4.0, >= 2.0
-Latest version of gollum: 4.1.4
-So is the gollum requirement outdated? false
-
-Benchmark with rubygems API 
-Rehearsal ----------------------------------------------
-             0.010000   0.000000   0.010000 (  1.126841)
-------------------------------------- total: 0.010000sec
-
-                 user     system      total        real
-             0.000000   0.000000   0.000000 (  0.672203)
-```
+* These results are stored as JSON in Redis using a `Redis::HashKey`. The default Redis key under which this `HashKey` is saved is `:github`.
+  
+
+## Example usage
+
+See `test.rb`. The basics:
+
+* start a Redis server with `redis-server`
```

Comment:
Can these steps be automated?

Reply:
Redis needs to be setup separately like any database (although I found it far simpler to get running than e.g. MySQL). The only way we could avoid the need for a database application is to use SQLite, but then a) Sidekiq wouldn't work b) E.g. DelayedJob could work but we wouldn't be able to multithread.

Reply:
I understand, but it would be nice if there was one script to fire up both sinatra and the redis server (or check if the server isn't already running).

Reply:
I don't see a way of doing this that is guaranteed to work even for Rails, but of course we could write a script that calls `redis-service` -- that just wouldn't be great on systems where you'd rather run redis as a service, or people who already have a server running.